doc/source/tutorial-quickstart-tensorflow.rst

```diff
@@ -79,93 +94,101 @@ implemented in the following way:
             return loss, len(x_test), {"accuracy": float(accuracy)}
 
 
-We can now create an instance of our class :code:`CifarClient` and add one line
-to actually run this client:
+Next, we create a client function that returns instances of :code:`CifarClient` on-demand when called:
+
+.. code-block:: python
+
+    def client_fn(cid: str):
+        return CifarClient().to_client()
+
+Finally, we create a :code:`ClientApp()` object that uses this client function:
 
 .. code-block:: python
 
-     fl.client.start_client(server_address="[::]:8080", client=CifarClient().to_client())
+    app = ClientApp(client_fn=client_fn)
 
 
 That's it for the client. We only have to implement :code:`Client` or
-:code:`NumPyClient` and call :code:`fl.client.start_client()`. If you implement a client of type :code:`NumPyClient` you'll need to first call its :code:`to_client()` method. The string :code:`"[::]:8080"` tells the client which server to connect to. In our case we can run the server and the client on the same machine, therefore we use
-:code:`"[::]:8080"`. If we run a truly federated workload with the server and
-clients running on different machines, all that needs to change is the
-:code:`server_address` we point the client at.
+:code:`NumPyClient`, create a :code:`ClientApp`, and pass the client function to it. If we implement a client of type :code:`NumPyClient` we'll need to first call its :code:`to_client()` method.
 
 
 Flower Server
 -------------
 
-For simple workloads we can start a Flower server and leave all the
+For simple workloads, we create a :code:`ServerApp` and leave all the
 configuration possibilities at their default values. In a file named
-:code:`server.py`, import Flower and start the server:
+:code:`server.py`, import Flower and create a :code:`ServerApp`:
 
 .. code-block:: python
 
-    import flwr as fl
+    from flwr.server import ServerApp
 
-    fl.server.start_server(config=fl.server.ServerConfig(num_rounds=3))
+    app = ServerApp()
 
 
 Train the model, federated!
 ---------------------------
 
 With both client and server ready, we can now run everything and see federated
-learning in action. FL systems usually have a server and multiple clients. We
-therefore have to start the server first:
+learning in action. First, we run the :code:`flower-superlink` command in one terminal to start the infrastructure. This step only needs to be run once.
```

Comment:
Before doing that, should we have a short section on local `flwr run` with the simulation engine?